### PR TITLE
Add Hopenhayn lecture

### PR DESCRIPTION
**Pull request by kp992**
This PR adds a new lecture on The Hopenhayn Entry-Exit Model.

Related to #168

### CODE REVIEW

File: lectures/_toc.yml [hunk 1]
```diff
@@ -18,6 +18,12 @@ parts:
   chapters:
   - file: lucas_model
   - file: markov_asset
+- caption: Macroeconomic Models
```
Comment (jstac) [inline lectures/_toc.yml:21]: Please put this section after "Dynamic Programming" rather than before.

File: lectures/hopenhayn.md [hunk 2]
```diff
@@ -0,0 +1,871 @@
...
+The Hopenhayn (1992, ECMA) entry-exit model is a highly influential (partial equilibrium) heterogeneous agent model where 
+
+* the agents receiving idiosyncratic shocks are firms, and
+* the model produces a non-trivial firm size distribution and a range of predictions for firm dynamics.
+
+We are going to study an extension of the basic model that has unbounded productivity.
+
+Among other things, this allows us to match the heavy tail observed in the firm size distribution data.
+
+The code for computing the equilibrium will be written in Python/JAX.
```
Comment (jstac) [inline lectures/hopenhayn.md:30]: Please delete this sentence.

File: lectures/hopenhayn.md [hunk 3]
```diff
@@ -0,0 +1,871 @@
...
+y = [eccdf(x) for x in x_grid]
+
+fix, ax = plt.subplots()
+ax.loglog(x_grid, y, 'o', label="ECCDF")
+ax.set_xlabel("productivity")
+ax.set_ylabel("counter CDF")
+plt.show()
+```
+
+**Exercise**
```
Comment (jstac) [inline lectures/hopenhayn.md:791]: Please put this in an exercise environment, and the solution below in a solution environment.

**Comment by jstac** [comment]
Thanks @kp992 , much appreciated.  Could you please make the minor requested changes?

-

File: lectures/hopenhayn.md (hunk 2 above)
Comment (jstac) [reply lectures/hopenhayn.md:30]: (The sentence starting "The code for...")

**Comment by kp992** [comment]
Thanks for the review @jstac. I have fixed them.

**Comment by jstac** [comment]
Perfect.  Many thanks @kp992

**Pull request merged**